splash/network_manager.py

```diff
@@ -177,6 +178,13 @@ def _handle_custom_proxies(self, request):
         if hasattr(request, 'custom_proxy'):
             self.setProxy(request.custom_proxy)
 
+            user, password = request.custom_proxy.user(), request.custom_proxy.password()
+            if not user and password:
```

Comment:
Note that this is evaluated as `if (not user) and password:` so if there is no user and no password still sets the header